.github/workflows/test.yml

```diff
@@ -0,0 +1,37 @@
+name: Unit Tests
+
+on:
+  pull_request:
+  push:
+    paths-ignore:
+      - '*.md'
+    branches:
+      - master
+      - develop
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+
+      - name: Run unit tests
+        run: go test -v -coverprofile=coverage.out -covermode=atomic ./...
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2
```

Comment:
I'm kind of not a fan of bots spamming PRs with coverage updates, does this action adds a comment to the PR? 🥲 

If so, maybe this could be added as check in the PR and when you click to the details button, it'll show the coverage?

If there is no a ready to use GH Action for this, would you be comfortable removing it from here?